apps/client/src/hoc/withChecker.js

```diff
@@ -1,29 +1,32 @@
 import React, { useState, useEffect, useContext } from "react";
 import { Redirect } from "react-router-dom";
-import Context from "../context";
+
+import { SessionContext, CheckerContext } from "../context";
 import { routes, geturl } from "../routes";
 import withAddress from "./withAddress";
+import getChecker from "../sttr_client";
 import LoadingPage from "../pages/LoadingPage";
 import ErrorPage from "../pages/ErrorPage";
-import getChecker from "../sttr_client";
 
 const dir =
   process.env.REACT_APP_STTR_ENV === "production" ? "PROD" : "STAGING";
 
 const withChecker = (Component) =>
   withAddress(({ ...rest }) => {
-    const context = useContext(Context);
-    const [checker, setChecker] = useState(context.checker);
+    const sessionContext = useContext(SessionContext);
+    const checkerContext = useContext(CheckerContext);
+    const [checker, setChecker] = useState(checkerContext.checker);
     const [error, setError] = useState();
 
     const { topic } = rest;
-    if (!context.address) {
+    if (!sessionContext.address) {
```

Comment:
But sessionContext.address is actually an Object right? Shouldn't you check for `[topic.slug]` as well?

Reply:
We shouldn't need this check right? It's in the withAddress let's check it there.

Reply:
@afjlambert `sessionContext.address` is either `null` or an object with data. If `sessionContext.address` exists it means that the address has been set. Bu as Sven says, it's already been placed in the `withAddress`